pkg/tests/end_to_end_tests/trace_query_integration_test.go

```diff
@@ -67,15 +63,22 @@ func TestCompareTraceQueryResponse(t *testing.T) {
 		err = ingestor.IngestTraces(context.Background(), copyTraces(sampleTraces))
 		require.NoError(t, err)
 
-		// Create Promscale proxy to query traces data.
-		promscaleProxy, err := jaegerquery.New(pgxconn.NewPgxConn(db), telemetry.NewNoopEngine())
+		// Start Promscale's HTTP endpoint for Jaeger query.
+		router, _, err := buildRouter(db)
 		require.NoError(t, err)
 
-		shutdown := runPromscaleTraceQueryJSONServer(promscaleProxy, ":9203")
-		defer shutdown.Close()
+		err = promscaleJaeger.ExtendQueryAPIs(router, pgxconn.NewPgxConn(db), telemetry.NewNoopEngine())
+		require.NoError(t, err)
 
-		promscaleClient := httpClient{"http://localhost:9203"}
+		go func() {
+			listener, err := net.Listen("tcp", ":9203")
```

Comment:
Shouldn't we use Promscale default port since router is registered under Promscale endpoint